cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/http/AbstractHttpHandlerDelegator.java

```diff
@@ -118,11 +140,60 @@ protected final DelayedHttpServiceResponder wrapResponder(HttpResponder responde
   protected final BodyConsumer wrapContentConsumer(HttpContentConsumer consumer,
                                                    DelayedHttpServiceResponder responder,
                                                    TransactionContext txContext) {
+    Preconditions.checkState(!responder.hasBufferedResponse(),
+                             "Response cannot be sent when using HttpContentConsumer.");
```

Comment:
Perhaps "HttpContentConsumer may not be used after a response has already been sent"